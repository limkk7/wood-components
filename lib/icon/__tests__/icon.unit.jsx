import Icon from '../icon'
import * as renderer from 'react-test-renderer'
import * as React from 'react'
import { mount} from 'enzyme'

describe('icon', () => {
  it('render successfully', () => {
    const json = renderer.create(<Icon name="qq"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('on Click', () => {
    let n = 1;
    const fn = () => {
      n = 2
    }
    const component = mount(<Icon name="wood" onClick={fn} />)
    component.find('svg').simulate('click')
    expect(n).toEqual(2)
  })
  it('on Click', () => {
    const fn = jest.fn()
    const fn2 = jest.fn()

    const component = mount(<Icon name="wood" onClick={fn} />)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})