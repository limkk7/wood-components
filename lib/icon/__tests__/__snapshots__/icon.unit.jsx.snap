// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="w-ui-icon"
  >
    <use
      xlinkHref="#qq"
    />
  </svg>
</span>
`;
